libgeopmd/src/StatsCollector.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2015 - 2024 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "StatsCollector.hpp"
+
+#include "geopm_stats_collector.h"
+
+#include <sstream>
+#include <cmath>
+#include <cstring>
+
+#include "geopm/Exception.hpp"
+#include "geopm/Helper.hpp"
+#include "geopm/PlatformIO.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm_time.h"
+#include "RuntimeStats.hpp"
+
+namespace geopm
+{
+
+    StatsCollector::StatsCollector()
+        : StatsCollector(std::vector<geopm_request_s> {})
+    {
+
+    }
+
+    StatsCollector::StatsCollector(const std::vector<geopm_request_s> &requests)
+        : StatsCollector(requests, platform_io())
+    {
+
+    }
+
+    StatsCollector::StatsCollector(const std::vector<geopm_request_s> &requests, PlatformIO &pio)
+        : m_pio(pio)
+        , m_stats(std::make_shared<RuntimeStats>(register_requests(requests)))
+        , m_time_begin(geopm::time_curr_string())
```

Comment:
First in constructor

m_time_idx = pio.push_signal("TIME", 0, 0)

At first update(): get real time and subtract pio.sample(m_time_idx) and store this as m_time_begin.  

Also do not report end time, but rather, provide seconds since start time in the report yaml by calling pio.sample(m_time_idx) at report generation time.